src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -326,25 +338,55 @@ public async Task<GetTransactionModel> GetTransactionAsync(string txid)
             return model;
         }
 
+        [ActionName("listaddressgroupings")]
+        [ActionDescription("Returns a list of grouped addresses which have had their common ownership made public by common use as inputs or as the resulting change in past transactions.")]
+        public AddressGroupingModel[] ListAddressGroupings()
+        {
+            if (!this.nodeSettings.TxIndex)
+                throw new RPCServerException(RPCErrorCode.RPC_INVALID_REQUEST, $"{nameof(ListAddressGroupings)} is incompatible with transaction indexing turned off (i.e. -txIndex=0).");
+
+            var addressGroupings = this.walletManager.GetAddressGroupings();
+            var addressGroupingModels = new List<AddressGroupingModel>();
+
+            foreach (var addressGrouping in addressGroupings)
+            {
+                var addressGroupingModel = new AddressGroupingModel();
+
+                foreach (var address in addressGrouping)
+                {
+                    var balance = this.walletManager.GetAddressBalance(address);
+                    addressGroupingModel.AddressGroups.Add(new AddressGroupModel()
+                    {
+                        Address = address,
+                        Amount = balance.AmountConfirmed
```

Comment:
Just a note, Core has some interesting logic around how this amount is computed for each address:

```
            int nDepth = pcoin->GetDepthInMainChain(locked_chain);
            if (nDepth < (pcoin->IsFromMe(ISMINE_ALL) ? 0 : 1))
                continue;
```

`(wallet.cpp, GetAddressBalances())`

i.e. the required number of confirmations seems to be dependent on the transaction origin.

Reply:
@zeptin is something that I should investigate in this PR?